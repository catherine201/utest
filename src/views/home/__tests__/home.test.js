import React from 'react';
// import renderer from 'react-test-renderer';  // 这个不支持antdesign
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import home from '..';

Enzyme.configure({ adapter: new Adapter() });
describe('(Component) home -- snapshot', () => {
  it('capture snapshot of home', () => {
    const footer = shallow(<home />);
    expect(footer).toMatchSnapshot();
  });
});
