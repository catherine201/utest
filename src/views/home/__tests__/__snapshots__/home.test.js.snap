// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AppSider -- snapshot capture snapshot of AppSider 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <img
            alt="logo"
            src="logo.png"
          />
        </div>,
        <div
          style={
            Object {
              "margin": "10px 0",
            }
          }
        >
          <h1>
            Euen 后台管理系统
          </h1>
        </div>,
        <div>
          <Button
            block={false}
            className="btn-block btn-lg"
            ghost={false}
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          >
            进入系统...
          </Button>
        </div>,
      ],
      "className": "middle-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo"
            src="logo.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Euen 后台管理系统
          </h1>,
          "style": Object {
            "margin": "10px 0",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Euen 后台管理系统",
          },
          "ref": null,
          "rendered": "Euen 后台管理系统",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            block={false}
            className="btn-block btn-lg"
            ghost={false}
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          >
            进入系统...
          </Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "进入系统...",
            "className": "btn-block btn-lg",
            "ghost": false,
            "loading": false,
            "onClick": [Function],
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": "进入系统...",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <img
              alt="logo"
              src="logo.png"
            />
          </div>,
          <div
            style={
              Object {
                "margin": "10px 0",
              }
            }
          >
            <h1>
              Euen 后台管理系统
            </h1>
          </div>,
          <div>
            <Button
              block={false}
              className="btn-block btn-lg"
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            >
              进入系统...
            </Button>
          </div>,
        ],
        "className": "middle-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              src="logo.png"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Euen 后台管理系统
            </h1>,
            "style": Object {
              "margin": "10px 0",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Euen 后台管理系统",
            },
            "ref": null,
            "rendered": "Euen 后台管理系统",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              block={false}
              className="btn-block btn-lg"
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            >
              进入系统...
            </Button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "进入系统...",
              "className": "btn-block btn-lg",
              "ghost": false,
              "loading": false,
              "onClick": [Function],
              "prefixCls": "ant-btn",
            },
            "ref": null,
            "rendered": "进入系统...",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`(Component) home -- snapshot capture snapshot of home 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "home",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "home",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
